src/app/modules/shared_models/keypair_model.nim

```diff
@@ -37,6 +39,48 @@ QtObject:
       [{i}]:({$self.items[i]})
       """
 
+  proc getSelectedKeyPair*(self: KeyPairModel): QVariant {.slot.} =
+    if self.selectedItem.isNil:
+      return newQVariant()
+    return newQVariant(self.selectedItem)
+
+  proc selectedKeyPairChanged(self: KeyPairModel) {.signal.}
+
+  QtProperty[QVariant] selectedKeyPair:
```

Comment:
@Cuteivist we don't need to know anything about selected key pair, that's not of our interest here.

Reply:
I agree... a model should be just a model, no logic preferably.